src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -291,7 +291,7 @@ def on_save_checkpoint(self) -> Dict:
 
     def on_load_checkpoint(self, state_dict: Dict) -> None:
         # cache the dataloader state dict until the dataloader objects are available
-        self._dataloader_state_dict = state_dict.get("dataloader_state_dict")
+        self._dataloader_state_dict = state_dict["dataloader_state_dict"]
```

Comment:
`self._dataloader_state_dict` is assumed to be `Dict[str, Any]`. If this is correct, we should use `[]` over `get()` to remove the unnecessary `Optional`.

🔗 https://github.com/Lightning-AI/lightning/pull/13555/files#diff-43414ae0ccfd0bccd332b194b079094c9aba0bcc75ac32e5c1c71cbac2b75cc1L74